great_expectations/cli/suite.py

```diff
@@ -116,15 +182,52 @@ def suite_new(ctx, expectation_suite, interactive, profile, batch_request, no_ju
         )
         sys.exit(1)
 
-    _suite_new_workflow(
-        context=context,
-        expectation_suite_name=expectation_suite,
-        interactive=interactive,
-        profile=profile,
-        no_jupyter=no_jupyter,
-        usage_event=usage_event_end,
-        batch_request=batch_request,
-    )
+    # TODO: AJB 20210420 add better warning if user passes --non-interactive but either --profile or --batch-request (for this and suite_edit...)
+
+    # If user has provided a flag determining their configuration, skip prompt.
+    if (interactive is not None) or (profile is True) or (batch_request is not None):
+        # Assume batch needed if user passes --profile
+        if profile is True and (interactive is False or interactive is None):
+            cli_message(
+                "<green>Entering interactive mode since you passed the --profile flag</green>"
+            )
+            interactive = True
+        # Assume batch needed if user passes --batch-request
+        elif (batch_request is not None) and (
+            interactive is False or interactive is None
+        ):
+            cli_message(
+                "<green>Entering interactive mode since you passed the --batch-request flag</green>"
+            )
+            interactive = True
+    else:
+        suite_create_method = click.prompt(
+            """\
+How would you like to create your Expectation Suite?
+    1. Manually, without interacting with a sample batch of data (default)
+    2. Interactively, with a sample batch of data
+    3. Automatically, using a profiler
+""",
+            type=click.Choice(["1", "2", "3"]),
+            show_choices=False,
+            default="1",
+            show_default=False,
+        )
+        # Default option
+        if suite_create_method == "":
+            interactive = False
+            profile = False
+        elif suite_create_method == "1":
+            interactive = False
+            profile = False
+        elif suite_create_method == "2":
+            interactive = True
+            profile = False
+        elif suite_create_method == "3":
+            interactive = True
+            profile = True
+
+    return {"interactive": interactive, "profile": profile}
```

Comment:
Nice! 👍

Reply:
Thanks! I thought about returning a tuple but I like having names for returned values esp when more than one.